docs/poly-apis.md

```diff
@@ -0,0 +1,636 @@
+
+*   `script`: [`Script`](#type-script) - Script's deploy info
+
+
+### Type `Script`
+
+Describes the lock script and type script for a cell.
+
+#### Examples
+
+
+```
+{
+  "code_hash": "0x28e83a1277d48add8e72fadaa9248559e1b632bab2bd60b27955ebc4c03800a5",
+  "hash_type": "data",
+  "args": "0x"
+}
+```
+
+
+#### Fields
+
+`Script` is a JSON object with the following fields.
+
+*   `code_hash`: [`H256`](#type-h256) - The hash used to match the script code.
+
+*   `hash_type`: [`ScriptHashType`](#type-scripthashtype) - Specifies how to use the `code_hash` to match the script code.
+
+*   `args`: [`JsonBytes`](#type-jsonbytes) - Arguments for script.
+
+
+### Type `ScriptHashType`
+
+Specifies how the script `code_hash` is used to match the script code and how to run the code.
+
+Allowed kinds: "data", "type" and "data1".
+
+Refer to the section [Code Locating](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#code-locating) and [Upgradable Script](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#upgradable-script) in the RFC *CKB Transaction Structure*.
+
+`ScriptHashType` is equivalent to `"data" | "type" | "data1"`.
+
+*   Type "data" matches script code via cell data hash, and run the script code in v0 CKB VM.
+*   Type "type" matches script code via cell type script hash.
+*   Type "data" matches script code via cell data hash, and run the script code in v1 CKB VM.
+
+
+### Type `JsonBytes`
+
+Variable-length binary encoded as a 0x-prefixed hex string in JSON.
+
+#### Example
+
+|  JSON | Binary |
+| --- |--- |
+|  “0x” | Empty binary |
+|  “0x00” | Single byte 0 |
+|  “0x636b62” | 3 bytes, UTF-8 encoding of ckb |
+|  “00” | Invalid, 0x is required |
+|  “0x0” | Invalid, each byte requires 2 digits |
+
+### Type `Versions`
+
+Version info of Godwoken & Web3 & Web3 Indexer.
+
+#### Examples
+
+
+```
+{
+  "web3Version": "1.0.0-rc1",
+  "web3IndexerVersion": "1.0.0-rc1",
+  "godwokenVersion": "1.1.0 f3cdd47"
+}
+```
+
+
+#### Fields
+
+`Versions` is a JSON object with the following fields.
+
+*   `web3Version`: `string` - Version of Godwoken Web3
+
+*   `web3IndexerVersion`: `string` - Version of Godwoken Web3 Indexer
+
+*   `godwokenVersion`: `string` - Version of Godwoken
+
+
+### Type `RollupCell`
+
+#### Fields
+
+`RollupCell` is a JSON object with the following fields.
+
+*   `typeHash`: [`H256`](#type-h256) - Hash of typeScript
+
+*   `typeScript`: [`Script`](#type-script) - Deploy info of rollup cell
+
+
+### Type `RollupConfig`
+
+#### Fields
+
+`RollupConfig` is a JSON object with the following fields.
+
+*   `requiredStakingCapacity`: [`Uint64`](#type-uint64) - Stacking capacity
+
+*   `challengeMaturityBlocks`: [`Uint64`](#type-uint64) - Challenge maturity blocks
+
+*   `finalityBlocks`: [`Uint64`](#type-uint64) - Finality Blocks
+
+*   `chainId`: [`Uint64`](#type-uint64) - Chain ID
```

Comment:
A useful reference for me: https://eips.ethereum.org/EIPS/eip-155
I prefer to add this link to the type description.

Reply:
Updated